rust/lance/src/dataset/scanner.rs

```diff
@@ -658,41 +659,33 @@ impl Scanner {
             ));
         }
         // make sure the field exists
-        let field = self
-            .dataset
-            .schema()
-            .field(column)
-            .ok_or(Error::invalid_input(
-                format!("Column {} not found", column),
+        let (_, element_type) = get_vector_type(self.dataset.schema(), column)?;
+        let dim = get_vector_dim(self.dataset.schema(), column)?;
+        // make sure the query is valid
+        if q.len() % dim != 0 {
```

Comment:
for vectors, `q.len() == dim`
for multivectors, `q.len() % dim == 0`

Reply:
i noticed our async API has supported query with batch vectors, so passing a list of vectors as query is now for that.
so querying with a multivector is by passing a flatten array

Reply:
Lets not design API like this. From users, it is confusing / got them by surprise, that they need use flatten array.

We can detect this route to use `multi-vector query + the column is list<fsl>` ?

Reply:
unfortunately, we can't. for multivector column, the query can be any number of vectors. say `query = [vec1, vec2, ..., vec_n]` can mean both:
1. n queries, each with single vector
2. single query with n vectors

Reply:
Could you explain more how those two are different?